tests/cross_functional/scale/noobaa/test_delete_objects.py

```diff
@@ -190,3 +201,96 @@ def test_delete_objects(
+        ],
+    )
+    def test_delete_objects_with_expiration(
+        self,
+        is_expiration,
+        bucket_factory,
+        reduce_expiration_interval,
+        scale_noobaa_resources_session,
+        change_lifecycle_schedule_min,
+        change_lifecycle_batch_size,
+        awscli_pod_session,
+        test_directory_setup,
+        mcg_obj_session,
+    ):
+        """
+        This test aims to test deletion of objects through normal recursive and expiration
+        method. then verify that the all the objects in the bucket are marked with deletion
+        time.
+
+        """
+
+        # reduce expiration interval to 1 minute
+        reduce_expiration_interval(interval=1)
+        log.info("Reduced expiration interval to 1 minute")
+
+        # change lifecycle schedule minutes
+        change_lifecycle_schedule_min(interval=1)
+        log.info("Change lifecycle schedyle minute to 1 minute")
+
+        # change lifecycle batch size to 10K to enable faster deletion
+        change_lifecycle_batch_size(new_lifecycle_batch_size=10000)
+        log.info("Increased the lifecycle batch size to 10K")
+
+        # generate 1 million empty files with unique identifiers
+        generate_empty_files(
+            awscli_pod_session,
+            dir=test_directory_setup.origin_dir,
+            amount=1000000,
+            timeout=3600,
+        )
+
+        # create the bucket
+        bucket = bucket_factory()[0]
+        log.info(f"Created bucket {bucket.name}")
+
+        # sync all objects generated above to the bucket
+        sync_object_directory(
+            awscli_pod_session,
+            test_directory_setup.origin_dir,
+            f"s3://{bucket.name}",
+            mcg_obj_session,
+            timeout=10800,
+        )
+        log.info(f"Uploaded objects to the bucket {bucket.name}")
+
+        if is_expiration:
+            # change the creation time for the objects in the bucket
+            expire_objects_in_bucket(bucket.name)
+            log.info(
+                f"Changed creation date for the objects in the bucket {bucket.name}"
+            )
+
+            # apply the object expiration policy to the bucket
+            log.info(f"Setting object expiration on bucket: {bucket.name}")
+            lifecycle_policy = LifecyclePolicy(
+                ExpirationRule(days=1, filter=LifecycleFilter())
+            )
+            mcg_obj_session.s3_client.put_bucket_lifecycle_configuration(
+                Bucket=bucket.name, LifecycleConfiguration=lifecycle_policy.as_dict()
+            )
+        else:
+            # remove the objects in the bucket recursively
+            rm_object_recursive(
+                awscli_pod_session, bucket.name, mcg_obj_session, timeout=7200
+            )
+            log.info("Deleted objects from the bucket recursively")
+
+        # Verify that all the objects are marked as deleted
+        verify_objs_deleted_from_objmds(bucket.name, timeout=72000, sleep=90)
```

Comment:
20hr timeout is too much. why do we need to have a 20hr timeout? 
Is there anything that we can do to have a reasonable timeout?

Reply:
As discussed in the weekly syncup, the only way to go forward is with this long timeout. About 1000 objects gets expired every 1 to 2 minutes timespan. Given 1 million object, 1000000/1000=1000 * 1.5 = 1500/60=25 hours. As per my two last runs we need atleast 20 hour for all the objects to expire. Currently we dont have support to increase the object expiration batch size above 1000.

Reply:
ok, this test is going to be into scale suite. Let's get feedback from scale team on this.

Reply:
25 hours for single test seems too much. We should find alternative way to reduce it. CC: @ramkiperiy

Reply:
@sagihirshfeld So i was under the same impression that we can use that env variable to modify the batch size. But as per my discussion with Ben we cant modify to a size greater than 1000, you can only modify the size within the limit of 1000 objects. It will work if you set the value 900 but wont work for 1500.
Bug: https://issues.redhat.com/browse/DFBUGS-472

Reply:
+1, I would suggest to reduce the amt of objs, since its limitation in MCG and why we should consider the limitation in CI coverage. 
There is already concerns raised to reduce the runtime of Noobaa scale tests which @udaysk23 is looking for enhancements.
If we would like to have this part of CI scripts then I would suggest to have some scale_extended tier marker and don't add it part of scale tier, with scale_extended tier marker you can execute the particular TC on a need basis and it won't be executed part of production or regression suit runs.
@mashetty330 @PrasadDesala @hnallurv ^^

Reply:
I have moved long running tests to different PR. will address it seperately